src/main/java/org/betonquest/betonquest/quest/condition/advancement/AdvancementConditionFactory.java

```diff
@@ -0,0 +1,43 @@
+package org.betonquest.betonquest.quest.condition.advancement;
+
+import org.betonquest.betonquest.Instruction;
+import org.betonquest.betonquest.api.quest.condition.PlayerCondition;
+import org.betonquest.betonquest.api.quest.condition.PlayerConditionFactory;
+import org.betonquest.betonquest.exceptions.InstructionParseException;
+import org.betonquest.betonquest.quest.PrimaryServerThreadData;
+import org.betonquest.betonquest.quest.condition.PrimaryServerThreadPlayerCondition;
+import org.betonquest.betonquest.utils.Utils;
+import org.bukkit.Bukkit;
+import org.bukkit.NamespacedKey;
+import org.bukkit.advancement.Advancement;
+
+/**
+ * Factory to create advancement conditions from {@link Instruction}s.
+ */
+public class AdvancementConditionFactory implements PlayerConditionFactory {
+    /**
+     * Data used for condition check on the primary server thread.
+     */
+    private final PrimaryServerThreadData data;
+
+    /**
+     * Create the Advancement Condition Factory.
+     *
+     * @param data the data used for checking the condition on the main thread
+     */
+    public AdvancementConditionFactory(final PrimaryServerThreadData data) {
+        this.data = data;
+    }
+
+    @Override
+    public PlayerCondition parsePlayer(final Instruction instruction) throws InstructionParseException {
+        final String advancementString = instruction.next();
+        if (!advancementString.contains(":")) {
+            throw new InstructionParseException("The advancement '" + advancementString + "' is missing a namespace!");
+        }
+        final String[] split = advancementString.split(":");
```

Comment:
I would consider refactoring this in a way that we first split, and then check if it two parts. I know it was this way before, so you can also leave it like it is now

Reply:
We should make the code better when we touch it.